### PR TITLE
Documentation Clarification on torch.compile Example

### LINKED ISSUE #110917: Documentation Clarification on torch.compile Example
### 📚 The doc issue

I came across an example in the [torch.compiler documentation](https://pytorch.org/docs/main/torch.compiler_get_started.html)

```py
import torch
def fn(x, y):
    a = torch.cos(x).cuda()
    b = torch.sin(y).cuda()
    return a + b
new_fn = torch.compile(fn, backend="inductor")
input_tensor = torch.randn(10000).to(device="cuda:0")
a = new_fn(input_tensor, input_tensor)
```

The documentation mentions that *we can turn 2 reads and 2 writes into 1 read and 1 write*. However, given there are three distinct operations, I'm inclined to think there should be 4 reads (x, y, a, b) and 3 writes (a, b, a+b). Can you confirm this?

Additionally, there's a note which says:

*To run this script, you need to have at least one GPU on your machine. If you do not have a GPU, you can remove the cuda() code in the snippet below and it will run on CPU.*

However, the crucial step to ensure GPU usage is the `.to(device="cuda:0")`. Merely removing .cuda() won't make the tensor reside on the CPU. 

### Suggest a potential alternative/fix

Hence, I'd suggest updating the note to:

```note
To run this script, you need to have at least one GPU on your machine. If you do not have a GPU, you can remove the `.to(device="cuda:0")` code in the snippet below and it will run on CPU.
```

Lastly, since .cuda() in the fn function seems redundant if we're already specifying the device using .to(device="cuda:0"), consider updating the function as:

```py
import torch
def fn(x, y):
    a = torch.cos(x)
    b = torch.sin(y)
    return a + b
```

Thank you for considering these suggestions. Looking forward to your feedback!

cc @svekars @carljparker

### PR DESCRIPTION
**Pull request by yewentao256**
Fixes #110917

### CODE REVIEW

File: docs/source/torch.compiler_get_started.rst [hunk 1]
```diff
@@ -33,7 +33,8 @@ be something like ``torch.relu()``. Pointwise ops in eager mode are
 suboptimal because each one would need to read a tensor from the
 memory, make some changes, and then write back those changes. The single
 most important optimization that inductor performs is fusion. In the
-example above we can turn 2 reads and 2 writes into 1 read and 1 write which
+example above we can turn 4 reads (``x``, ``y``, ``a``, ``b``) and
```
Comment (msaroufim) [inline docs/source/torch.compiler_get_started.rst:36]: I think we can make this example clearer with something like

```python
def fn(x):
  a = torch.cos(x) 
  b = torch.sin(a)
  return b
```

So here you have 1 read `x` and a single write `b` as opposed to read `x`, write `a`, read `a`, write `b`

**Review by msaroufim — CHANGES_REQUESTED** [review]
see comment

File: docs/source/torch.compiler_get_started.rst (hunk 1 above)
Comment (yewentao256) [reply docs/source/torch.compiler_get_started.rst]: Great suggestion! I've made the necessary updates to the code. Could you please review it again?

**Comment by msaroufim** [comment]
@yewentao256 lemme know when you're done fighting CI and I can give a final stamp and merge. Thanks!

**Comment by yewentao256** [comment]
> @yewentao256 lemme know when you're done fighting CI and I can give a final stamp and merge. Thanks!

@msaroufim It appears that all the CI checks have passed, with the exception of a few that were cancelled. Could you please rerun the CI? I don't have the permissions to do so. Thanks!

**Review by msaroufim — APPROVED** [review]

**Comment by yewentao256** [comment]
@msaroufim Hello, I've noticed that lots of CI jobs have been cancelled. Do you have any idea why this might be happening?

**Comment by msaroufim** [comment]
Not sure what's going on @malfet any idea?

**Comment by malfet** [comment]
Rebase cancels jobs in progress on previous version of the PR, but then it should schedule new ones, unless there is a GitHub outage

File: docs/source/torch.compiler_get_started.rst [hunk 2]
```diff
@@ -14,26 +14,27 @@ understanding of how you can use ``torch.compile`` in your own programs.
...
 .. code:: python
 
    import torch
-   def fn(x, y):
-       a = torch.cos(x).cuda()
-       b = torch.sin(y).cuda()
-       return a + b
+   def fn(x):
+      a = torch.cos(x)
+      b = torch.sin(a)
```
Comment (malfet) [inline docs/source/torch.compiler_get_started.rst:25]:
```suggestion
      b = torch.sin(a)
```

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**